### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by minextu**
See commit messages.
closes https://github.com/archzfs/archzfs/issues/75, closes https://github.com/archzfs/archzfs/issues/47, closes https://github.com/archzfs/archzfs/issues/88

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -256,7 +260,7 @@ fi
 
 msg "$(date) :: ${script_name} started..."
 
-
+get_headers_conflicts
```
Comment (demizer) [inline build.sh]: Fantastic!

File: src/kernels/linux-hardened.sh [hunk 2]
```diff
@@ -64,9 +64,9 @@ update_linux_hardened_pkgbuilds() {
 update_linux_hardened_git_pkgbuilds() {
     pkg_list=("spl-linux-hardened-git" "zfs-linux-hardened-git")
     kernel_version=${kernel_version_git}
-    kernel_version_full=$(kernel_version_full ${kernel_version_git})
-    kernel_version_full_pkgver=$(kernel_version_full_no_hyphen ${kernel_version_git})
-    kernel_version_major=${kernel_version_git%-*}
+    kernel_version_full=$(kernel_version_full ${kernel_version})
+    kernel_version_full_pkgver=$(kernel_version_full_no_hyphen ${kernel_version})
+    kernel_version_major=${kernel_version%-*}
```
Comment (demizer) [inline src/kernels/linux-hardened.sh]: I am uncertain how this change will affect the build scripts. But I do want to take a deeper look at this. For now I will hold off on merging these changes until #139 is ready to be worked on.

Comment (minextu) [inline src/kernels/linux-hardened.sh]: This does not change the pkgver,  `kernel_version` is set to `kernel_version_git` on top:
`kernel_version=${kernel_version_git}`
This way you can copy any instructions made in the none git functions, and don't have to change every occurrence of `kernel_version`. It's only a minor change

Comment (demizer) [inline src/kernels/linux-hardened.sh]: Ah I see, I misinterpreted it then. Thanks!

**Pull request merged**